Add documentation of existing tests

hug-dev opened the pull request:
Adds a new page in our book to describe the tests that we have on a
high-level view.
Close #35

ionut-arm commented on src/parsec_service/tests/existing_tests.md:
```diff
@@ -0,0 +1,55 @@
+# List of existing tests
+
+## Integration tests on providers
+
+| Name                                                                                          | Description                                                                                                                                                                                                                                                                             |
+|-----------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [TPM integration tests](https://github.com/parallaxsecond/parsec/tree/master/tests/providers) | These tests initialize a provider structure and perform operation using the `Provide` trait. A software crypto library, like `ring`, can be used to verify the behaviour. [Example with the TPM provider](https://github.com/parallaxsecond/parsec/blob/master/tests/providers/tpm.rs). |
+
```
Sorry to bug you with this, but I'm moving these to the E2E tests 😬  we could do it either way - you change this or  change it after mine gets merged.

ionut-arm replied:
I guess you can just say in the "Normal tests" line below that we also verify Parsec results with 3rd party software implementations of crypto (`ring`, the `rsa` crate...)

ionut-arm commented on src/parsec_service/tests/existing_tests.md:
```diff
@@ -0,0 +1,55 @@
+
+| Name                                                                                                                                     | Description                                                                                                       |
+|------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| [All providers e2e tests](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/all_providers)                            | End-to-end tests on a service with all providers. Mostly to test core operations.                                 |
```
You could also mention that now we do "cross testing" - either in this line or in another one, where we test that the output of various operations from different providers is accepted by the other providers.

ionut-arm reviewed (commented):
Thanks, this covers our tests quite nicely!

hug-dev commented:
Thanks for the review! Addressed all of your points. The CI link check will fail on accessing `cross.rs` but will reload once your other PR is merged 🚀

ionut-arm reviewed (approved):
👌
